src/Ampersand/PatchHelper/Helper/PatchOverrideValidator.php

```diff
@@ -147,69 +111,156 @@ public function canValidate($file)
+
+    /**
+     * @return array
+     */
+    public function getErrors()
+    {
+        return array_filter($this->errors);
     }
 
     /**
      * Use the object manager to check for preferences
-     *
-     * @param string $file
-     * @throws \Exception
      */
-    private function validatePhpFile($file)
+    private function validatePhpFileForPreferences()
     {
+        $file = $this->appCodeFilepath;
+
         $class = ltrim($file, 'app/code/');
         $class = preg_replace('/\\.[^.\\s]{3,4}$/', '', $class);
         $class = str_replace('/', '\\', $class);
 
         $preferences = [];
 
-        foreach (array_keys($this->areaConfig) as $area) {
-            if (isset($this->areaConfig[$area]['preferences'][$class])) {
-                $preference = $this->areaConfig[$area]['preferences'][$class];
+        $areaConfig = $this->m2->getAreaConfig();
+        foreach (array_keys($areaConfig) as $area) {
+            if (isset($areaConfig[$area]['preferences'][$class])) {
+                $preference = $areaConfig[$area]['preferences'][$class];
                 if ($this->isThirdPartyPreference($class, $preference)) {
                     $preferences[] = $preference;
                 }
             }
         }
 
         // Use raw framework
-        $preference = $this->config->getPreference($class);
+        $preference = $this->m2->getConfig()->getPreference($class);
         if ($this->isThirdPartyPreference($class, $preference)) {
             $preferences[] = $preference;
         }
 
         $preferences = array_unique($preferences);
 
-        if (!empty($preferences)) {
-            $exception = new ClassPreferenceException();
-            $exception->setFilePaths($preferences);
-            throw $exception;
+        foreach ($preferences as $preference) {
+            $this->errors[self::TYPE_PREFERENCE][] = $preference;
+        }
+    }
+
+    /**
+     * Check for plugins on modified methods within this class
+     */
+    private function validatePhpFileForPlugins()
+    {
+        $file = $this->appCodeFilepath;
+
+        $class = ltrim($file, 'app/code/');
+        $class = preg_replace('/\\.[^.\\s]{3,4}$/', '', $class);
+        $class = str_replace('/', '\\', $class);
+
+        /*
+         * Collect a list of non-magento plugins on the given class
+         */
+        $nonMagentoPlugins = [];
+
+        $areaConfig = $this->m2->getAreaConfig();
+        foreach (array_keys($areaConfig) as $area) {
+            $tmpClass = $class;
+            if (!isset($areaConfig[$area][$tmpClass]['plugins'])) {
+                //Search with and without the preceding slash
+                $tmpClass = "\\$tmpClass";
+            }
+            if (isset($areaConfig[$area][$tmpClass]['plugins'])) {
+                foreach ($areaConfig[$area][$tmpClass]['plugins'] as $pluginName => $pluginConf) {
+                    $pluginClass = $pluginConf['instance'];
+                    $pluginClass = rtrim($pluginClass, '\\');
```

Comment:
I think this `rtrim` needs to become an `ltrim`, could it be?

I saw some classes it reported which started with a `\`. For consistency's sake it might be nicer to display them all the same without a starting `\`, which I think was the purpose here?

Reply:
This should definitely be a `ltrim` thanks for the code review.

Agreed, I think what I was trying to do here was ensure that if people plugged into `\Magento\Class` and `Magento\Class` that it would pick up both of them? I'd need to write a test to ensure I've got it done properly though.

Same idea applies to the output too.